### PR TITLE
Add `#[derive(Clone, Copy...` on all bitflags

### PR DESCRIPTION
**Pull request by nyurik**
This will make it easier to use them, e.g. I won't need to re-create `FunctionFlags` for registering multiple similar functions.

### CODE REVIEW

**Comment by gwenn** [comment]
`Eq` and `PartialEq` seem useless

**Comment by nyurik** [comment]
They are clearly not as useful as `Copy`, agree, but I am sure there could be usecases where users will want to pass them around and compare them, wouldn't they?  Also, there could be some implementers that wrap your library, and they might need to take extra actions before passing bitfield into the inner lib.   But this is up to you - my main req is to copy bitfields to pass them around more easily (e.g. have a global const/static, etc).

Speaking of derive - there are many enums that also do not implement `Copy`, e.g. `types::Type` should be both `Copy` and `Hash` (someone may want to have a lookup for type -> value conversion of sorts?).  Do you have any overall rule of thumb of what gets which derives? Thx!

**Comment by gwenn** [comment]
>  I am sure there could be usecases where ...

YAGNI: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

**Comment by nyurik** [comment]
@gwenn YAGNII is my fav principle too, but the cost of submitting a PR and discussing it, and waiting for release is also pretty high, so what's the middle ground?  I will add the clone/copy where makes sense - that's what I do need. Is that ok?

**Comment by gwenn** [comment]
> I will add the clone/copy where makes sense - that's what I do need. Is that ok?

Yes, please.

**Comment by nyurik** [comment]
i tried to clean them up a bit, adding just the copy/clone where it seem to make sense.  Overall, looking at many `pub enum` cases, it seems the derives are very unorganized/ad-hoc, without much consistency.  I will try to tame my OCD though :)

**Comment by nyurik** [comment]
@gwenn should i make any changes to this PR?

File: src/vtab/mod.rs [hunk 1]
```diff
@@ -322,7 +323,7 @@ pub trait UpdateVTab<'vtab>: CreateVTab<'vtab> {
 
 /// Index constraint operator.
 /// See [Virtual Table Constraint Operator Codes](https://sqlite.org/c3ref/c_index_constraint_eq.html) for details.
-#[derive(Debug, Eq, PartialEq)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```
Comment (gwenn) [inline src/vtab/mod.rs:326]: Created only by SQLite itself (readonly on rusqlite side)

File: src/vtab/mod.rs [hunk 2]
```diff
@@ -59,6 +59,7 @@ use crate::{str_to_cstring, Connection, Error, InnerConnection, Result};
 // ffi::sqlite3_vtab_cursor => VTabCursor
 
 /// Virtual table kind
+#[derive(Copy, Clone, Debug)]
```
Comment (gwenn) [inline src/vtab/mod.rs:62]: Used only as a constant at compile time.

File: src/vtab/series.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ const SERIES_COLUMN_STOP: c_int = 2;
 const SERIES_COLUMN_STEP: c_int = 3;
 
 bitflags::bitflags! {
-    #[derive(Clone, Copy)]
+    #[derive(Clone, Copy, Debug)]
```
Comment (gwenn) [inline src/vtab/series.rs:31]: Use only to pass information between `best_index` and `filter` methods.

File: src/session.rs [hunk 4]
```diff
@@ -655,7 +655,7 @@ impl Connection {
 /// See [here](https://sqlite.org/session.html#SQLITE_CHANGESET_CONFLICT) for details.
 #[allow(missing_docs)]
 #[repr(i32)]
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```
Comment (gwenn) [inline src/session.rs:658]: Created only by SQLite (readonly on rusqlite side)

File: src/vtab/mod.rs (hunk 1 above)
Comment (nyurik) [reply src/vtab/mod.rs]: reverted

File: src/vtab/mod.rs (hunk 2 above)
Comment (nyurik) [reply src/vtab/mod.rs]: reverted

File: src/session.rs (hunk 4 above)
Comment (nyurik) [reply src/session.rs]: reverted

File: src/vtab/series.rs (hunk 3 above)
Comment (nyurik) [reply src/vtab/series.rs]: reverted

**Review by nyurik — COMMENTED** [review]
thx, reverted the ones you suggested, and also merged main for ease of review.  You may want to squash when merging though

**Pull request merged**